Makefile.dep

```diff
@@ -787,6 +787,14 @@ ifneq (,$(filter periph_gpio_irq,$(USEMODULE)))
   FEATURES_REQUIRED += periph_gpio
 endif
 
+ifneq (,$(RIOTBOOT_LEN))
+  FEATURES_PROVIDED += riotboot
```

Comment:
This is wrong, boards (boards common) should say `FEATURES_PROVIDED += riotboot`. It would remove the need to whitelist them.

Also, should it not be somehow a `slot` feature more than `riotboot` ? or both maybe.

Reply:
The idea here is that if a board defined a RIOTBOOT_LEN, the board specific riotboot variables are present, thus the board supports the riotboot feature. If RIOTBOOT_LEN is not set, there's no riotboot for the board.

Reply:
Currently it's defined in `cortexm_common/Makefile.include` which could be contested but there is not changes request on it yet. And using `cortexm_common` is not enough to support it. You need to use `cortexm_common/ldscript/cortexm.ld`, define the `ROM_ADDR`, … variables and have a cpu_init that can be called multiple times. The current configuration also currently does not support having an integrated bootloader or using `nordic_softdevice_ble`.

Also, it's a definition that would go basically in a `Makefile.include`, which is not included when parsing dependencies by `info-boards-supported` (and should not be). It is working right now when doing make as we parse them in a wrong order https://github.com/RIOT-OS/RIOT/issues/9913.

This shows me that `FEATURES_REQUIRED += riotboot` is missing in `tests/riotboot` or as a dependency of `slot_util` maybe.

```
git diff .
diff --git a/tests/riotboot/Makefile b/tests/riotboot/Makefile
index 3e49c4f0b..d0bcfcadb 100644
--- a/tests/riotboot/Makefile
+++ b/tests/riotboot/Makefile
@@ -10,6 +10,8 @@ BOARD_WHITELIST += samr21-xpro iotlab-m3
 USEMODULE += slot_util
 USEMODULE += riot_hdr

+FEATURES_REQUIRED += riotboot
+
 # Comment this out to disable code in RIOT that does safety checking
 # which is not needed in a production environment but helps in the
 # development process:

$ make info-boards-supported

```